pkg/processor/reconciler_helper.go

```diff
@@ -49,7 +49,7 @@ func (h *ReconcilerHelper) BlockExists(
 	ctx context.Context,
 	block *types.BlockIdentifier,
 ) (bool, error) {
-	_, err := h.blockStorage.GetBlock(ctx, block)
+	_, err := h.blockStorage.GetBlock(ctx, types.ConstructPartialBlockIdentifier(block))
 	if err == nil {
```

Comment:
nit: feels like it's more idiomatic to evaluate `if err != nil` first
```
        _, err := h.blockStorage.GetBlock(ctx, types.ConstructPartialBlockIdentifier(block))
	if err != nil {
		if errors.Is(err, storage.ErrBlockNotFound) {
			return false, nil
		}

		return false, err
	}

	return true, nil
```

Reply:
Although I agree it is more idiomatic to do `if err != nil` first, I think it is more preferable to avoid nesting conditionals.